main.go

```diff
@@ -178,6 +179,22 @@ func main() {
 	}
 }
 
+func setupQueueManager(mgr ctrl.Manager, cCache *cache.Cache, cfg *configapi.Configuration) *queue.Manager {
+	queues := queue.NewManager(
+		mgr.GetClient(),
+		cCache,
+		queue.WithQueueVisibilityUpdateInterval(cfg.QueueVisibility.UpdateIntervalSeconds),
+		queue.WithQueueVisibilityClusterQueuesMaxCount(cfg.QueueVisibility.ClusterQueues.MaxCount),
+	)
+	if cfg.QueueVisibility.ClusterQueues.MaxCount != 0 {
```

Comment:
if maxcount = 0, the feature is disabled, and the snapshot is not computed. Here we add cluster manager as a dependency component to manager to be started when Start is called. The same we do with scheduler

Reply:
I see, makes sense, please add a comment above then. 

I was also thinking about the alternative to move the scheduling of taking snapshots to the clusterqueue_controller. Taking the snapshots is done per CQ so I have this intuition of putting the mechanism closer for CQ management. For example, I think CQ controller should know the `UpdateIntervalSeconds` to skip the update if the time since the lastChangeTime is less than  `UpdateIntervalSeconds`.

Reply:
added a comment above